packages/sdk/src/actions/getDeposit.ts

```diff
@@ -0,0 +1,230 @@
+import { Address, Hex, isHash, PublicClient } from "viem";
+import { getDepositFromLogs } from "./getDepositFromLogs";
+import { Deposit } from "../types";
+import { getFillByDepositTx } from "./getFillByDepositTx";
+import { NoFillLogError } from "../errors";
+
+export type GetDepositParams = {
+  originChainClient: PublicClient;
+  destinationChainClient: PublicClient;
+  findBy: {
+    originChainId: number;
+    destinationChainId: number;
+    destinationSpokePoolAddress: Address;
+  } & Partial<{
+    originSpokePoolAddress: Address;
+    depositId: number;
+    depositTxHash: Hex;
+  }>;
+  depositLogFromBlock?: bigint;
+  fillLogFromBlock?: bigint;
+  indexerUrl?: string;
+};
+
+export type GetDepositReturnType = Promise<Deposit>;
+
+/**
+ * Get a deposit by its deposit tx hash or deposit id + spoke pool address.
+ * @param params - See {@link GetDepositParams}.
+ * @returns See {@link Deposit}.
+ */
+export async function getDeposit(
+  params: GetDepositParams,
+): GetDepositReturnType {
+  const {
+    originChainClient,
+    destinationChainClient,
+    findBy,
+    depositLogFromBlock,
+    fillLogFromBlock,
+    indexerUrl,
+  } = params;
+
+  const { originChainId, destinationSpokePoolAddress } = findBy;
+
+  let rawDeposit: Deposit | undefined;
+
+  if (
+    !(
+      findBy.depositTxHash ||
+      (findBy.depositId && findBy.originSpokePoolAddress)
+    )
+  ) {
+    throw new Error(
+      "You must provide either a 'depositTxHash' or 'depositId' + 'spokePoolAddress'",
+    );
+  }
+
+  // Try to get deposit from logs by tx hash on origin chain
+  if (findBy.depositTxHash) {
+    const depositTxHash = findBy.depositTxHash;
+
+    if (!isHash(depositTxHash)) {
+      throw new Error(`Can't get receipt for tx hash: ${depositTxHash}`);
+    }
+
+    const receipt = await originChainClient.getTransactionReceipt({
+      hash: depositTxHash,
+    });
+
+    if (!receipt) {
+      throw new Error(`No receipt found for tx hash: ${depositTxHash}`);
+    }
+
+    rawDeposit = getDepositFromLogs({ originChainId, receipt });
+  }
+  // Try to get deposit from logs by deposit id and spoke pool address on origin chain
+  else if (findBy.depositId && findBy.originSpokePoolAddress) {
+    const { originSpokePoolAddress, depositId } = findBy;
+
+    const [depositEvent] = await originChainClient.getLogs({
+      address: originSpokePoolAddress,
+      event: {
+        anonymous: false,
+        inputs: [
+          {
+            indexed: false,
```

Comment:
We can import the abi from `SpokePool.ts`

Reply:
Yea, I tried that but it is a bit clunky... We only need the item and also have a const assertion to type inference to work